File: app/assets/javascripts/Quality.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const QualitySchema = DataElementSchema({
  type: Any,
  standardSequence: Any,
  start: Any,
  end: Any,
  score: Any,
  method: Any,
  truthTP: Any,
  queryTP: Any,
  truthFN: Any,
  queryFP: Any,
  gtFP: Any,
  precision: Any,
  recall: Any,
  fScore: Any,
  roc: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'Quality' },

});

module.exports.QualitySchema = QualitySchema;
module.exports.Quality = mongoose.model('Quality', QualitySchema);

File: app/models/qdm/quality.rb
module QDM
  # app/models/qdm/quality.rb
  class Quality < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :type
    field :standardSequence
    field :start
    field :end
    field :score
    field :method
    field :truthTP
    field :queryTP
    field :truthFN
    field :queryFP
    field :gtFP
    field :precision
    field :recall
    field :fScore
    field :roc
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
